### PR TITLE
[Draft] Add test cases for BPF_MAP_TYPE_HASH_OF_MAPS and BPF_MAP_TYPE_ARRAY _OF_MAPS

### PR DESCRIPTION
**Pull request by shpalani**
## Description

Add missing test cases for 

- Hash of Maps with ANY inner map type, without static initializer.
- Array of Maps with ANY inner map type, with static initializer.

outer map is BPF_MAP_TYPE_HASH_OF_MAPS, Inner map can by ANY map.
inner map is BPF_MAP_TYPE_ARRAY, Inner map can be ANY map.

## Testing

_Do any existing tests cover this change? Are new tests needed?_ 

## Documentation

_Is there any documentation impact for this change?_ No

## Installation

_Is there any installer impact for this change?_No

### CODE REVIEW

**Review by dthaler — DISMISSED** [review]

**Review by dthaler — DISMISSED** [review]

File: libs/execution_context/ebpf_native.c [hunk 1]
```diff
@@ -917,15 +922,16 @@ _ebpf_native_set_initial_map_values(_Inout_ ebpf_native_module_t* module)
 
             ebpf_handle_t handle_to_insert = ebpf_handle_invalid;
 
-            if (native_map_to_update->entry->definition.type == BPF_MAP_TYPE_ARRAY_OF_MAPS) {
+            ebpf_map_type_t map_type = native_map_to_update->entry->definition.type;
+            if (map_type == BPF_MAP_TYPE_ARRAY_OF_MAPS || map_type == BPF_MAP_TYPE_HASH_OF_MAPS) {
```
Comment (Alan-Jowett) [inline libs/execution_context/ebpf_native.c:926]: This doesn't make sense. Only array type maps can be statically initialized as they have an implicit key value. Hash style maps have no implicit key value and can't be statically initialized.

**Review by Alan-Jowett — CHANGES_REQUESTED** [review]
Please verify that statically initializing maps works on Linux and what the behavior is when they key value is not an integer.

File: libs/execution_context/ebpf_native.c (hunk 1 above)
Comment (shpalani) [reply libs/execution_context/ebpf_native.c:926]: Analysis in https://github.com/microsoft/ebpf-for-windows/issues/2882#issuecomment-1730680224

Comment (shpalani) [reply libs/execution_context/ebpf_native.c:926]: Thanks.  Agreed that the change in ebpf_native.c is not needed.
Then as discussed with @shankarseal, the **bpf2c** should flag the error that it cannot initialize the value of Hash-of-Maps. I will create a separate issue.


Create an issue to add the missing test cases.
- Hash of Maps with **ANY inner map type**, without static initializer.
- Array of Maps with **ANY inner map type**, with static initializer.

**Comment by shpalani** [comment]
Moving it to draft.

**Comment by Alan-Jowett** [comment]
@shpalani What is the status of this PR? Has it been abandoned? If so, please close it. If not, please address commends and move it to ready for review.

**Comment by dahavey** [comment]
@shpalani please reopen if you want to continue working on this?

**Pull request closed without merging**